src/gui/transferlistsortmodel.cpp

```diff
@@ -209,6 +215,7 @@ int TransferListSortModel::compare(const QModelIndex &left, const QModelIndex &r
 
     case TransferListModel::TR_DLLIMIT:
     case TransferListModel::TR_DLSPEED:
+    case TransferListModel::TR_PRIVATE:
```

Comment:
>There seems to be something wrong with the sorting

Damn it, we've overlooked it after all. This is the wrong place because it is eventually compared as an integer.
```c++
return customCompare(leftValue.toInt(), rightValue.toInt());
``` 
It should be moved outside of this group of cases and placed separately:
```c++
case TransferListModel::TR_PRIVATE:
    return customCompare(leftValue, rightValue);
```